mirgecom/initializers.py

```diff
@@ -462,6 +450,176 @@ def exact_rhs(self, discr, q, t=0.0):
+    expression in the :func:`~MulticomponentLump.exact_rhs` method.
+
+    .. automethod:: __init__
+    .. automethod:: __call__
+    .. automethod:: exact_rhs
+    """
+
+    def __init__(
+            self, *, dim=1, nspecies=0,
+            rho0=1.0, p0=1.0,
+            center=None, velocity=None,
+            spec_y0s=None, spec_amplitudes=None,
+            spec_centers=None
+    ):
+        r"""Initialize MulticomponentLump parameters.
+
+        Parameters
+        ----------
+        dim: int
+            specify the number of dimensions for the lump
+        rho0: float
+            specifies the value of $\rho_0$
+        p0: float
+            specifies the value of $p_0$
+        center: numpy.ndarray
+            center of lump, shape ``(dim,)``
+        velocity: numpy.ndarray
+            fixed flow velocity used for exact solution at t != 0,
+            shape ``(dim,)``
+        """
+        if center is None:
+            center = np.zeros(shape=(dim,))
+        if velocity is None:
+            velocity = np.zeros(shape=(dim,))
+        if len(center) != dim or len(velocity) != dim:
+            raise ValueError(f"Expected {dim}-dimensional vector inputs.")
+
+        if nspecies > 0:
+            if spec_y0s is None:
+                spec_y0s = np.ones(shape=(nspecies,))
+            if spec_centers is None:
+                spec_centers = make_obj_array([np.zeros(shape=dim,)
+                                               for i in range(nspecies)])
+            if spec_amplitudes is None:
+                spec_amplitudes = np.ones(shape=(nspecies,))
+            if len(spec_y0s) != nspecies or\
+               len(spec_amplitudes) != nspecies or\
+                   len(spec_centers) != nspecies:
+                raise ValueError(f"Expected nspecies={nspecies} inputs.")
+            for i in range(nspecies):
+                if len(spec_centers[i]) != dim:
+                    raise ValueError(f"Expected {dim}-dimensional "
+                                     f"inputs for spec_centers.")
+
+        self._nspecies = nspecies
+        self._dim = dim
+        self._velocity = velocity
+        self._center = center
+        self._p0 = p0
+        self._rho0 = rho0
+        self._spec_y0s = spec_y0s
+        self._spec_centers = spec_centers
+        self._spec_amplitudes = spec_amplitudes
+
+    def __call__(self, x_vec, *, t=0, eos=IdealSingleGas()):
+        """
+        Create a multi-component lump solution at time *t* and locations *x_vec*.
+
+        Note that *t* is used to advect the component lumps under the assumption of
+        constant, and uniform velocity.
+
+        Parameters
+        ----------
+        t: float
+            Current time at which the solution is desired
+        x_vec: numpy.ndarray
+            Nodal coordinates
+        eos: :class:`mirgecom.eos.GasEOS`
+            Equation of state class to be used in construction of soln (if needed)
```

Comment:
> if needed

It looks like it's always needed? What does this mean? And: do all EOSs have a `.gamma()`?

IOW: This interface doesn't seem well-specified.

Reply:
54230c7